services/order.py

```diff
@@ -0,0 +1,56 @@
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+from django.db import transaction
+from django.utils.datetime_safe import datetime
+
+from db.models import Order, User, Ticket
+
+
+def create_order(tickets: list, username: str, date: datetime = None) -> Order:
+    if not tickets:
+        raise ValueError("No tickets to create")
+
+    with transaction.atomic():
+        try:
+            user = User.objects.get(username=username)
+        except ObjectDoesNotExist:

```

Comment:
Consider using `User.DoesNotExist` instead of `ObjectDoesNotExist` for clarity and specificity in exception handling.

Reply:
Agree